naptime/src/main/scala/org/coursera/naptime/ari/engine/EngineImpl.scala

```diff
@@ -211,20 +211,23 @@ class EngineImpl @Inject() (
       requestHeader: RequestHeader,
       requestField: RequestField,
       data: Iterable[DataMap],
+      schema: RecordDataSchema,
       path: Seq[String],
       reverse: ReverseRelationAnnotation): Future[FieldRelationResponse] = {
     val resourceName = ResourceName.parse(reverse.resourceName).getOrElse {
       throw new IllegalStateException(s"Could not parse identifier " +
         s"'${reverse.resourceName}''")
     }
     val argumentsByElement = data.map { topLevelElement =>
-      val InterpolationRegex = new Regex("""\$([a-zA-Z0-9_]+)""", "variableName")
+      val InterpolationRegex = new Regex("""\$(?:([a-zA-Z0-9_]+)|\{([^\}]+)\})""", "withoutBraces", "withBraces")
       val arguments: Set[(String, JsValue)] = reverse.arguments
         .mapValues { value =>
           InterpolationRegex.replaceAllIn(
             value, { regexMatch =>
-              val variableName = regexMatch.group("variableName")
-              Option(topLevelElement.get(variableName)).map {
+              val withoutBraces = Option(regexMatch.group("withoutBraces"))
+              val withBraces = Option(regexMatch.group("withBraces"))
+              val variableName = withoutBraces.orElse(withBraces).getOrElse("")
```

Comment:
ah. i'll leave a comment. i want to match both:
`"$instructorIds"` and `"${instructorUnion/org.coursera.InstructorUnionType/instructorIds}"`